capnp/lib/capnp.pyx

```diff
@@ -3739,6 +3760,36 @@ cdef class _FlatArrayMessageReader(_MessageReader):
 
 
 @cython.internal
+cdef class _SegmentArrayMessageReader(_MessageReader):
+
+    cdef object _objects_to_pin
+
+    def __init__(self, segments):
+        # takes a Python array of bytes and constructs a ConstWordArrayArrayPtr
+        num_segments = len(segments)
+        cdef char* ptr
+        cdef schema_cpp.ConstWordArrayPtr seg_ptr
+        cdef schema_cpp.ConstWordArrayPtr* seg_ptrs = <schema_cpp.ConstWordArrayPtr*>malloc(num_segments * sizeof(schema_cpp.ConstWordArrayPtr))
```

Comment:
This is a memleak. You probably want to store this variable as a member of the class (below `_objects_to_pin`), and clean it up in __dealloc__.